src/state/contexts/TransactionState.tsx

```diff
@@ -0,0 +1,61 @@
+import { Dispatch, createContext, useContext, useReducer } from 'react';
+
+import {
+  AntInteraction,
+  CONTRACT_TYPES,
+  ContractType,
+  REGISTRY_INTERACTION_TYPES,
+  RegistryInteraction,
+} from '../../types';
+import { TransactionAction } from '../reducers/TransactionReducer';
+
+export type TransactionState = {
+  transactionData: { [x: string]: any }; // data that will be used to perform the transaction.
+  contractType: ContractType;
+  interactionType: AntInteraction | RegistryInteraction;
+  workflowStage: 'confirm' | 'deploy' | 'complete';
```

Comment:
'pending'?

Reply:
pending... confirmation to deploy or confirmation of deployment? "confirm" shows the confirmation stage, deploy shows the deployment stage and auto-procees to complete once deployed.

Reply:
i'd encourage we use terms that match the status of a tx on chain, not just what the component is showing. we can then do different things in the modal depending on these statuses

Reply:
Makes sense, added